crates/solver/src/solver/http_solver/settlement.rs

```diff
@@ -127,23 +151,47 @@ fn match_prepared_and_settled_orders(
 fn match_prepared_and_settled_amms(
     prepared_amms: Vec<Liquidity>,
     settled_amms: HashMap<usize, UpdatedAmmModel>,
-) -> Result<Vec<ExecutedAmm>> {
+) -> Result<Vec<(Option<ExecutionPlanCoordinatesModel>, Execution)>> {
```

Comment:
Rather than making this type more complex, how about implementing a getter for the coordinates on the Execution enum directly? This way you can just call `execution.coordinates()` everywhere. You could even implement `PartialOrd` on `Execution` which would make sorting even more simple (although I don't mind the explicitness and have been overly excited in the past to shoehorn logic into existing traits).